### DISCUSSION TITLE
Merge multiple config files

### DISCUSSION THREAD

**Discussion by adamkissvcc** [golangci/golangci-lint#2001] [Q&A]
Hello!

Is there a way, to use common configurations in a single file, and use a per-project override file for it? (Basically us a base config and an override config).

**Comment by butuzov** [comment]
I am using 3 approaches:

1) `--no-config`, in case if the project runs linter, but my colleagues didn't include config to the repository of the project. its remove conflict with mine, located at `~`/`home`

2) merging with[`yq`](https://github.com/mikefarah/yq) help, so I can have a bit straiter rules (but be careful).

```bash
# with change in place `.golangci.yaml` with a changes from `~/.golangci-lint.yaml`
yq ea -i 'select(fileIndex == 0) * select(fileIndex == 1)' .golangci.yaml ~/.golangci-lint.yaml
```
3) for testing proposes I am also using 2nd approach with combination modification operators of `yq` (e.g. `delete`)

```bash
yq ea -i 'del(.linters-settings.exhaustivestruct.struct-patterns)' .golangci.default.yaml
yq ea -i '.linters-settings.blocked.modules += "gopkg.in\/yaml.v2' .golangci.default.yaml
```
